src/QMCHamiltonians/DensityMatrices1B.cpp

```diff
@@ -576,6 +587,7 @@ void DensityMatrices1B::warmup_sampling()
 
 DensityMatrices1B::Return_t DensityMatrices1B::evaluate(ParticleSet& P)
 {
+  timers[DM_eval]->start();
```

Comment:
Please use RAII scoped timer.

Reply:
Please update your document ("recommended" -> "required when possible").  It currently lists both methods as options.